### PR TITLE
Add infrared YOLO baseline scripts and notes

### PR DESCRIPTION
**Pull request by Ming-912**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Adds infrared small-target baseline tooling with mask-to-YOLO label conversion, a reproducible training script, and companion notes for dataset setup, metrics, and code reading. 🚀

### 📊 Key Changes
- Adds `mask2yolo.py` to convert binary infrared segmentation masks into YOLO detection labels for `NUAA-SIRST` and `NUDT-SIRST`, including NUAA filename normalization.
- Adds `train_infrared.py` to create deterministic train/val splits, generate dataset config files, train a YOLO baseline, and log per-epoch `Ra`, `Fa`, `mAP50`, `mAP50-95`, and loss metrics.
- Saves custom evaluation output to `epoch_metrics_ra_fa.csv` inside the Ultralytics run directory for easier experiment tracking.
- Adds `INFRARED_BASELINE_NOTES.md` documenting dataset layout, usage, expected outputs, metric definitions, and baseline comparison results.
- Adds `YOLOv8_CODE_READING_NOTES.md` with a guided walkthrough of the training flow, model structure, loss, validation, and metrics for readers studying the implementation.

### 🎯 Purpose & Impact
- Makes it easier to reproduce and benchmark infrared small-target detection baselines from mask-based datasets. 📈
- Provides a practical starting point for researchers experimenting with custom Ra/Fa reporting alongside standard detection metrics.
- Improves onboarding for contributors by documenting both the workflow and relevant source files.
- Broadens repository utility for niche infrared detection research, though the additions are currently standalone scripts/docs rather than core package integrations.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Ming-912, thank you for submitting this `ultralytics/ultralytics` 🚀 PR for infrared YOLO baseline scripts and notes! This is an automated message to help with review prep, and an engineer will assist you shortly 😊 Please review the checklist below to help ensure a smooth integration:

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: mask2yolo.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+
+def mask_to_yolo_boxes(mask: np.ndarray) -> list[str]:
+    _, binary = cv2.threshold(mask, 0, 255, cv2.THRESH_BINARY)
+    contours, _ = cv2.findContours(binary, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
+
+    img_h, img_w = mask.shape[:2]
+    boxes = []
+    for contour in contours:
+        if cv2.contourArea(contour) < MIN_AREA:
```
Comment (UltralyticsAssistant) [inline mask2yolo.py:30]: ⚠️ **HIGH**: `cv2.contourArea()` is 0 for 1-pixel and some line-like contours, so with `MIN_AREA = 1` this drops the smallest valid targets entirely. On small-target infrared datasets that can silently erase ground-truth boxes and misreport masks as empty. If the goal is to preserve all annotated targets, filter on bounding-box area instead or remove this contour-area check.

File: train_infrared.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    results = model.train(
+        data=str(data_yaml),
+        epochs=args.epochs,
+        imgsz=args.imgsz,
+        batch=args.batch,
+        workers=args.workers,
+        device=args.device,
+        project=str(DEFAULT_PROJECT),
+        name=args.name,
+        exist_ok=True,
```
Comment (UltralyticsAssistant) [inline train_infrared.py:256]: 💡 **MEDIUM**: The script claims reproducible training, but only the split generation is seeded. The actual Ultralytics run still uses its default RNG state, so shuffling, augmentation, and PyTorch behavior can vary between runs. Pass the seed through to `model.train()` and enable deterministic mode if you want the same command to reproduce the same baseline.

**Suggested change:**
```suggestion
        exist_ok=True, seed=args.seed, deterministic=True,
```

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Good addition overall: the scripts are focused and the callback-based metric logging avoids patching core Ultralytics code. The main issues to address are preserving tiny mask instances during conversion and seeding the actual training run so the advertised reproducibility matches behavior.

💬 Posted 2 inline comments

**Comment by glenn-jocher** [comment]
Thanks for detailing this work; for future PRs, please keep changes tightly scoped to reusable core `ultralytics` functionality, avoid adding standalone research scripts/notes to the main repo, and include targeted tests plus any required docs updates per our [development workflow](https://handbook.ultralytics.com/workflows/development).
